src/test/java/com/khaliuk/serialization/ShapeToJsonTest.java

```diff
@@ -0,0 +1,112 @@
+package com.khaliuk.serialization;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class ShapeToJsonTest {
+    List<Shape> list1;
+    List<Shape> list2;
+    List<Shape> list3;
+    List<Shape> list4;
+    Shape group1;
+    Shape group2;
+    Shape group3;
+    Shape mainGroup;
+    ShapeToXml xmlEncoder;
+
+    @Before
+    public void setUp() {
+        list1 = new ArrayList<>();
+        list1.add(new Triangle(new double[]{3.0, 4.0, 5.0}));
+        list1.add(new Square(3.3));
+        list1.add(new Circle(2.5));
+        group1 = new Group(list1);
+
+        list2 = new ArrayList<>();
+        list2.add(new Triangle(new double[]{6.0, 8.0, 10.0}));
+        list2.add(new Square(3.7));
+        list2.add(new Circle(2.8));
+        list2.add(group1);
+        group2 = new Group(list2);
+
+        list3 = new ArrayList<>();
+        list3.add(new Triangle(new double[]{9.0, 5.0, 12.0}));
+        list3.add(new Square(5.3));
+        list3.add(new Circle(7.5));
+        group3 = new Group(list3);
+
+        list4 = new ArrayList<>();
+        list4.add(new Triangle(new double[]{1.5, 2.0, 2.5}));
+        list4.add(new Square(0.3));
+        list4.add(new Circle(2.25));
+        list4.add(group3);
+        list4.add(group2);
+        mainGroup = new Group(list4);
+    }
+
+    @Test
+    public void testEncodeReturnsProperFormattedJsonString() {
+        String expectedResult = "{\t\"Group\" : {\n" +
+                "\t\t\"Triangle\" : {\n" +
+                "\t\t\t\"side\"0 : \"1.5\",\n" +
```

Comment:
non correct json! inside Group you have List<Shape> => in json should be "shapes": [ ]